### PR TITLE
feat(session manager): Signal superblock beacon

### PR DESCRIPTION
**Pull request by girazoki**
This PR adds the superblock beacon to the lastbeacon message. In particular:

- Adds a new 'highest_superblock_checkpoint' to LastBeacon structure.
- If there exists some superblock state, adds the superblock checkpoint to the lastbeacon message.
- Else, signals the genesis hash and the 0 index.

The latter point should be modified when a decision is taken about how to bootstrap the network with an appropriate state #1367.

Currently upon reception of the superblock checkpoint the node does nothing. Once the consensus has been decided, we should implement how to handle that superblock state #1366.

### CODE REVIEW

File: data_structures/src/superblock.rs [hunk 1]
```diff
@@ -182,6 +183,23 @@ impl SuperBlockState {
             }
         }
     }
+
+    /// Produces a `SuperBlock` that includes the blocks in `block_headers` if there is at least one of them.
+    /// `ars_pkh_keys` will be used to validate all the superblock votes received for the
+    /// next superblock. The votes for the current superblock must be validated using
+    /// `previous_ars_identities`. The ordered bn256 keys will be merkelized and appended to the superblock
```
Comment (clbartoli) [inline data_structures/src/superblock.rs:190]: I think the description hast to be updated

**Comment by tmpolaczyk** [comment]
You should replace the `TODO` with `FIXME(#1355)`, or open new issues for some of the uncertainties, or fix them in this PR.

Other than that, this PR looks good to me.

File: data_structures/src/superblock.rs (hunk 1 above)
Comment (girazoki) [reply data_structures/src/superblock.rs:190]: very true haha

File: data_structures/src/superblock.rs [hunk 2]
```diff
@@ -182,6 +183,21 @@ impl SuperBlockState {
...
+        match (
+            self.current_superblock_index.is_some(),
+            self.current_superblock_hash.is_some(),
+        ) {
+            (true, true) => Some(CheckpointBeacon {
+                checkpoint: self.current_superblock_index?,
+                hash_prev_block: self.current_superblock_hash?,
+            }),
+            _ => None,
+        }
```
Comment (tmpolaczyk) [inline data_structures/src/superblock.rs:199]: Actually, this is all you need. The match is redundant
```suggestion
            Some(CheckpointBeacon {
                checkpoint: self.current_superblock_index?,
                hash_prev_block: self.current_superblock_hash?,
            })
```

File: node/src/actors/session/handlers.rs [hunk 3]
```diff
@@ -254,18 +254,30 @@ impl StreamHandler<BytesMut, Error> for Session {
...
                             highest_block_checkpoint,
+                            highest_superblock_checkpoint,
                         }),
                     ) => {
-                        session_last_beacon_inbound(self, ctx, highest_block_checkpoint);
+                        session_last_beacon_inbound(
+                            self,
+                            ctx,
+                            highest_block_checkpoint,
+                            highest_superblock_checkpoint,
```
Comment (tmpolaczyk) [inline node/src/actors/session/handlers.rs:264]: This looks redundant, `highest_block_checkpoint == highest_superblock_checkpoint`, right?

Comment (tmpolaczyk) [reply node/src/actors/session/handlers.rs:264]: Maybe you can change

                        Command::LastBeacon(LastBeacon {
                            highest_block_checkpoint,
                            highest_block_checkpoint,
                            highest_superblock_checkpoint,
                        }),

Into

    Command::LastBeacon(last_beacon)

And pass the two beacons at once:

```suggestion
                            last_beacon
```

File: node/src/actors/chain_manager/mod.rs [hunk 4]
```diff
@@ -75,6 +75,7 @@ use crate::{
     },
     signature_mngr, storage_mngr,
 };
+use witnet_data_structures::types::LastBeacon;
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs:78]: fmt

File: node/src/actors/chain_manager/mod.rs [hunk 5]
```diff
@@ -661,6 +662,18 @@ impl ChainManager {
             .highest_block_checkpoint
     }
 
+    /// Retrieves the latest superblock hash an index, if it exists. Else, returns the genesis beacon.
```
Comment (aesedepece) [inline node/src/actors/chain_manager/mod.rs:665]:
```suggestion
    /// Retrieves the latest superblock hash and its index, if it exists. Else, returns the genesis beacon.
```

**Pull request merged**